### PR TITLE
Make webpack aliases work in VS Code

### LINKED ISSUE #3952: [DX] Make webpack aliases work in VS Code
**Is your feature request related to a problem? Please describe.**
Webpack alias paths do not work in VS code - it is a pain to import files without autocomplete

**Describe the solution you'd like**
This is reportedly solvable using a JSConfig file - see links below.

**Describe alternatives you've considered**
None, but any other solution would be good too

**Testing notes**
[Tester to complete]

Dev insight: DX feature only - no testing needed

- [ ] This feature is expected to need manual testing.

**Additional context**
https://code.visualstudio.com/docs/languages/jsconfig
https://medium.com/@justintulk/solve-module-import-aliasing-for-webpack-jest-and-vscode-74007ce4adc9

### PR DESCRIPTION
**Pull request by FK78**
Resolves #3952

**Overall change:** _Added `jsconfig.json` file to fix webpack aliases._

**Code changes:**

- _Created a `jsconfig.json` file._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "es2017",
+    "allowSyntheticDefaultImports": false,
```
Comment (rhenshaw56) [inline jsconfig.json]: question, why are we setting `allowSyntheticDefaultImports` to false ??

**Review by rhenshaw56 — COMMENTED** [review]
do we really have to add `/index` to all the aliases ??

**Review by ryanmccombe — APPROVED** [review]
Looks good - thanks for doing this! 👍 

Do we need to disable the default imports for this to work?  Agree with Rowland's concerns

File: jsconfig.json (hunk 1 above)
Comment (FK78) [reply jsconfig.json]: Left that in by accident, removed now.

Comment (FK78) [reply jsconfig.json]: Left in by mistake, removed now.

File: src/app/containers/ATIAnalytics/index.jsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import AmpATIAnalytics from './amp';
 import getArticleAtiParams from './params/article';
 import getFrontPageAtiParams from './params/frontpage';
 import getMediaPageAtiParams from './params/media';
-import { pageDataPropType } from '#models/propTypes/data';
+import { pageDataPropType } from '#models/propTypes/data/index';
```
Comment (rhenshaw56) [inline src/app/containers/ATIAnalytics/index.jsx]: in this file, when I highlight and right-click on `pageDataPropType`  then select "Go to definition" it goes to the file where `pageDataPropType` was declared but when I do it for the other imports especially the relative ones, that doesn't seem to work @FK78 could you check this out ??

**Review by rhenshaw56 — CHANGES_REQUESTED** [review]

**Review by rhenshaw56 — APPROVED** [review]
Looks good!

**Review by dr3 — APPROVED** [review]

**Review by jroebu14 — APPROVED** [review]

File: webpack.config.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ module.exports = (shell = {}) => {
     mode: IS_PROD ? 'production' : 'development',
     devtool: IS_PROD ? 'source-map' : 'cheap-eval-source-map',
     resolve: {
-      extensions: ['.js', '.jsx'], // resolves `import '../Foo'` to `../Foo/index.jsx`
+      extensions: ['.js', '.jsx', '.json'], // resolves `import '../Foo'` to `../Foo/index.jsx`
```
Comment (dr3) [inline webpack.config.js]: I dont think this is needed, and risks bundling json files in the application

**Review by dr3 — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 3 above)
Comment (FK78) [reply webpack.config.js]: Talked IRL, reverted.

**Review by dr3 — APPROVED** [review]
I dont use vscode, but if everyone else is happy, I am happy with this change :)

**Review by thekp — COMMENTED** [review]
👌

**Pull request merged**